pkg/util/config.go

```diff
@@ -0,0 +1,72 @@
+package util
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/gcfg.v1"
+	vsphere "k8s.io/cloud-provider-vsphere/pkg/common/config"
+	"k8s.io/klog/v2"
+	legacy "k8s.io/legacy-cloud-providers/vsphere"
+)
+
+// VSphereConfig contains configuration for cloud provider.  It wraps the legacy version and the newer upstream version
+// with yaml support
+type VSphereConfig struct {
+	Config       *vsphere.Config
+	LegacyConfig *legacy.VSphereConfig
+}
+
+// LoadConfig load the desired config into this config object.
+func (c *VSphereConfig) LoadConfig(data string) error {
+	var err error
+	c.Config, err = vsphere.ReadConfig([]byte(data))
+	if err != nil {
+		return err
+	}
+
+	// Load legacy if cfgString is not yaml.  May be needed in areas that require legacy logic.
+	lCfg := &legacy.VSphereConfig{}
+	err = gcfg.ReadStringInto(lCfg, data)
+	if err != nil {
+		// For now, we can just log an info so that we know.
+		klog.V(2).Info("Unable to load cloud config as legacy ini.")
```

Comment:
I can include it, but its most likely from YAML.  If INI format itself is bad, then there would be other issues where the config object wouldn't load to begin with before loading the legacy info and this would never get executed.

Reply:
I updated this log message to contain error.